server/admin.go

```diff
@@ -668,21 +668,31 @@ func (s *adminServer) Drain(ctx context.Context, req *serverpb.DrainRequest) (*s
 
 	nowOn, err := s.server.Drain(on)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	nowOnInts := make([]int32, len(nowOn))
+	res := serverpb.DrainResponse{
+		On: make([]int32, len(nowOn)),
+	}
 	for i := range nowOn {
-		nowOnInts[i] = int32(nowOn[i])
+		res.On[i] = int32(nowOn[i])
+	}
+	if err := stream.Send(&res); err != nil {
+		return err
 	}
+
 	if req.Shutdown {
-		s.server.stopper.Quiesce()
-		go func() {
-			time.Sleep(50 * time.Millisecond)
-			s.server.stopper.Stop()
-		}()
+		go s.server.stopper.Stop()
+	}
+
+	ctx := stream.Context()
+
+	select {
+	case <-s.server.stopper.IsStopped():
```

Comment:
All those things happen already after the stopper is `Quiesce()`d and `Wait()`ed, so yes, there's a race here, but all the stopper is doing before process death is:
```
	for _, c := range s.closers {
		c.Close()
	}
	close(s.stopped)
```

There's a race here, but I just don't see how we can do any better (short of adding yet another phase to the stopper so that connections are the last thing to go).